File: WORDS1.cpp
#include <iostream>
#include <string>
using namespace std;
int main(){
	int n,m,i,j,l;
	bool flag;
	cin >> n ;
	string str;
	int arr[127];
	while(n--){
		for(i=97;i<127;i++) arr[i]=0;
		cin >> m ;
		while(m-->=0){
			getline(cin,str);
			arr[str[0]]++;
			arr[str[str.length()-1]]--;
		}
		j=0;
		l=0;
		flag = false;
		for(i=97;i<123;i++){
          if(arr[i] <-1 || arr[i] >1){
          	flag=true;
          	break;
          }else if(arr[i]==1) j++;
          else if(arr[i]==-1) l++;
		}
		if(flag) cout <<"\nThe door cannot be opened.";
		else{
			if((j==0 && l==0) || (j==1 && l==1))
				cout <<"\nOrdering is possible.";
			else cout <<"\nThe door cannot be opened.";
		}
	}
	cout <<endl;
	return 0;
}

File: WORDS1.py
num = int(raw_input())
for i in range(num):
	lst= [0]*26
	num_ = int(raw_input())
	for j in raw_input(num_):
		str_= raw_input()
		lst[ord(str_[0])-97]+=1
		lst[ord(str_[-1])-97]-=1
		start, end = 0,0
		flag=True;
		for k in range(26):
			if lst[k]>1 or lst[k] < -1:
				flag=False;
				break;
			elif lst[k]==1:
				start+=1
			elif lst[k]==-1:
				end+=1
		if flag==True:
			if (start==0 and end==0) or (start==1 and end==1):
				print "Ordering is possible."
			else:
				print"The door cannot be opened."
		else:
			print"The door cannot be opened."


